src/suffer.cpp

```diff
@@ -944,65 +1068,41 @@ void suffer::from_sunburn( Character &you )
-        you.mod_pain( 1 );
-        // Check exposure of all body parts
-        for( const std::pair<const bodypart_id, float> &bp_exp : bp_exposure ) {
-            const bodypart_id &this_part = bp_exp.first;
-            const float exposure = bp_exp.second;
-            // Skip parts with adequate protection
-            if( exposure <= MIN_EXPOSURE ) {
-                continue;
-            }
-            // Don't damage eyes directly, since it takes from head HP (in other words, your head
-            // won't be destroyed if only your eyes are exposed).
-            if( this_part == bodypart_id( "eyes" ) ) {
-                continue;
-            }
-            // Exposure percentage determines likelihood of injury
-            // 10% exposure is 10% chance of injury, naked = 100% chance
-            if( x_in_y( exposure, 1.0 ) ) {
-                // Because hands and feet share an HP pool with arms and legs, and the mouth shares
-                // an HP pool with the head, those parts take an unfair share of damage in relation
-                // to the torso, which only has one part.  Increase torso damage to balance this.
-                if( this_part == bodypart_id( "torso" ) ) {
-                    you.apply_damage( nullptr, this_part, 2 );
-                } else {
-                    you.apply_damage( nullptr, this_part, 1 );
-                }
-            }
-        }
-    } else {
-        // Albinism/datura causes pain (1/60) or focus loss (59/60)
-        if( one_turn_in( 1_minutes ) ) {
-            you.mod_pain( 1 );
-        } else {
-            you.mod_focus( -1 );
-        }
+
+    auto warn_and_wake_up = [ &you, &message, all_parts_list]( game_message_type type ) {
+        you.add_msg_if_player( type, message, all_parts_list );
+        // Wake up from skin irritation/burning
+        if( you.has_effect( effect_sleep ) ) {
+            you.wake_up();
+        }
+    };
+
+    //~ %s is a list of body parts.  The plurality integer is the total
+    //~ number of body parts
+    switch( worst_effect ) {
+        case Damage:
+            message = n_gettext( "Your %s is bathed in sunlight. It feels like it is burning up.",
+                                 "Your %s are bathed in sunlight. They feel like they are burning up.",
+                                 affected_part_names.size() );
+            warn_and_wake_up( m_bad );
+            break;
+        case Pain:
+            message = n_gettext( "The sunlight burns on your %s.",
+                                 "The sunlight burns on your %s.",
+                                 affected_part_names.size() );
+            warn_and_wake_up( m_bad );
+            break;
+        case Focus_Loss:
+            message = n_gettext( "The sunlight on your %s irritates you.",
+                                 "The sunlight on your %s irritates you.",
+                                 affected_part_names.size() );
+            warn_and_wake_up( m_bad );
```

Comment:
Should we use a different message type here? Red seems a bit aggressive for some focus loss. Actually this might not even wake up the character.

Reply:
It's a negative effect so red fits.